File: android/app/src/main/java/voidawt/Image.java
package voidawt;

import voidawt.image.ImageObserver;
import voidawt.image.ImageProducer;

public abstract class Image {
    public abstract int getWidth(ImageObserver observer);

    public abstract int getHeight(ImageObserver observer);

    public Graphics getGraphics() {
        return null;
    }

    public void flush() {
    }

    public ImageProducer getSource() {
        return null;
    }

    public abstract int[] peekArgb();
}

File: src/commonMain/kotlin/awt/Image.kt
package awt

import awt.image.ImageObserver

expect abstract class Image {
    fun flush()
    fun getScaledInstance(width: Int, height: Int, hints: Int): Image
    fun setAccelerationPriority(priority: Float)
    fun getAccelerationPriority(): Float
}

expect fun Image.getGraphics(): Graphics
expect fun Image.getWidth(observer: ImageObserver?): Int
expect fun Image.getHeight(observer: ImageObserver?): Int

expect val Image.pxWidth: Int
expect val Image.pxHeight: Int

expect val IMAGE_SCALE_DEFAULT: Int
expect val IMAGE_SCALE_SMOOTH: Int